odoo_backup_sh/controllers/main.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+import jinja2
+import os
+import requests
+import random
+import string
+import odoo
+from odoo import http
+from odoo.http import request
+from odoo.tools import config
+from odoo.addons import web
+
+path = os.path.realpath(os.path.join(os.path.dirname(__file__), '..', 'views'))
+loader = jinja2.FileSystemLoader(path)
+env = jinja2.Environment(loader=loader, autoescape=True)
+
+
+class BackupDatabase(web.controllers.main.Database):
+
+    def _render_template(self, **d):
+        res = super(BackupDatabase, self)._render_template(**d)
+        # Show button 'Restore via Odoo-backup.sh' on web/database/manager and web/database/selector pages
+        place_for_backup_button = res.find(
+            '<button type="button" data-toggle="modal" data-target=".o_database_restore"'
+        )
+        if place_for_backup_button == -1:
+            place_for_backup_button = res.find(
+                '<a role="button" data-toggle="modal" data-target=".o_database_restore" class="btn btn-link">'
+            )
+        if place_for_backup_button != -1:
+            d['list_db'] = config['list_db']
+            d['databases'] = []
+            try:
+                d['databases'] = http.db_list()
+            except odoo.exceptions.AccessDenied:
+                monodb = http.db_monodb()
+                if monodb:
+                    d['databases'] = [monodb]
```

Comment:
Try to get those variables via ``res.qcontext`` (when you need them or before first call of ``find``)

Reply:
def _render_template(self, **d) returns a rendered template as unicode string (see http://jinja.pocoo.org/docs/2.10/api/). There is no the qcontext attribute.

Reply:
Ok